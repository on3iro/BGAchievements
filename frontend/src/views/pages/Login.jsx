// @flow

import * as React from 'react'
import styled from 'styled-components'

import PageWrapper from 'views/components/atoms/PageWrapper'
import Content from 'views/components/atoms/Content'
import P from 'views/components/atoms/P'
import Link from 'views/components/atoms/Link'
import Divider from 'views/components/atoms/Divider'

import Header from 'views/components/organisms/Header'
import LoginForm from 'views/components/organisms/LoginForm'

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: ${props => (props.theme.buildingUnit * 54)}px;
`

class Login extends React.PureComponent<Object> {
  render () {
    return (
      <PageWrapper>
        <Header title='BGAchievements' />
        <Content>
          <Wrapper>
            <LoginForm />
            <P alignment='right'>
              <Link to='/forgot-password'>Forgot password</Link>
            </P>
            <Divider />
            <P alignment='center'>
              You don't have an account yet?<br />
              <Link to='/registration'>Register here</Link>
            </P>
          </Wrapper>
        </Content>
      </PageWrapper>
    )
  }
}

export {
  Login as default
}
