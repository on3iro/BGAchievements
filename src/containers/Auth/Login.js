import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import Button from 'components/Button'

import LoginForm from './LoginForm'
import FormDivider from './FormDivider'
import Wrapper from './Wrapper'

import { loginSubmit } from './ducks/actions'
import { getLoadingState } from './ducks/selectors'

export class Login extends Component {
  handleSubmit = values => {
    this.props.loginSubmit(values)
  }

  render () {
    return (
      <Wrapper>
        <LoginForm isLoading={this.props.isLoading} onSubmit={this.handleSubmit} />
        <FormDivider>Noch keinen Account?</FormDivider>
        <Button to='/register' secondary>Registrieren</Button>
      </Wrapper>
    )
  }
}

Login.propTypes = {
  isLoading: PropTypes.bool,
  loginSubmit: PropTypes.func.isRequired
}

export const mapStateToProps = state => {
  return {
    isLoading: getLoadingState(state)
  }
}

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    loginSubmit
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)
