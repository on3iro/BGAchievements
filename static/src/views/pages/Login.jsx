// @flow

import React from 'react'
import styled from 'styled-components'

import P from 'views/components/atoms/P'
import Link from 'views/components/atoms/Link'
import Divider from 'views/components/atoms/Divider'
import LoginForm from 'views/components/organisms/LoginForm'

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: ${props => (props.theme.buildingUnit * 54)}px;
`

const CenteredText = styled(P)`
  text-align: center;
`

const RightAlignedText = styled(P)`
  text-align: right;
`

const Login = () => (
  <Wrapper>
    <LoginForm />
    <RightAlignedText>
      <Link to='/home'>Forgot password</Link>
    </RightAlignedText>
    <Divider />
    <CenteredText>
      You don't have an account yet?<br />
      <Link to='/home'>Register here</Link>
    </CenteredText>
  </Wrapper>
)

export {
  Login as default
}
