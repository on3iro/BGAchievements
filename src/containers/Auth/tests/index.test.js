import React from 'react'
import { shallow } from 'enzyme'
import { Redirect } from 'react-router-dom'

import { Auth, mapStateToProps } from '../index'
import Login from '../Login'

const renderWrapper = newProps => {
  const props = {
    isLoggedIn: false,
    location: {
      state: 'test'
    },
    ...newProps
  }

  const renderedWrapper = shallow(<Auth {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<Auth />', () => {
  it('should render <Login />', () => {
    const { renderedWrapper } = renderWrapper()
    expect(renderedWrapper.find(Login).length).toEqual(1)
  })

  it('should render <Redirect /> when user is logged in', () => {
    const { renderedWrapper } = renderWrapper({isLoggedIn: true})
    expect(renderedWrapper.find(Redirect).length).toBe(1)
  })

  describe('react-redux', () => {
    it('should map state to props', () => {
      expect(mapStateToProps({
        auth: { user: { isLoggedIn: true }}
      })).toMatchSnapshot()
    })
  })
})
