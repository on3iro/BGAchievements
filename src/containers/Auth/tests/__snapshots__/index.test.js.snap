exports[`<Auth /> react-redux should map state to props 1`] = `
Object {
  "isLoggedIn": true,
}
`
